convert2rhel/unit_tests/actions/system_checks/package_updates_test.py

```diff
@@ -65,22 +79,150 @@ def test_check_package_updates(pretend_os, packages, exception, expected, monkey
+        package_updates_action,
+        level="OVERRIDABLE",
+        id="OUT_OF_DATE_PACKAGES",
+        message=(
+            "The system has 2 package(s) not updated based on the enabled system repositories.\n"
+            "List of packages to update: package-1 package-2.\n\n"
+            "Not updating the packages may cause the conversion to fail.\n"
+            "Consider updating the packages before proceeding with the conversion."
+        ),
+    )
+    assert (
+        "The system has 2 package(s) not updated based on the enabled system repositories.\n"
+        in caplog.records[-1].message
+    )
+
+
+@centos8
+def test_check_package_updates_not_up_to_date_skip(pretend_os, monkeypatch, package_updates_action, caplog):
+    packages = ["package-1", "package-2"]
+    monkeypatch.setattr(package_updates, "get_total_packages_to_update", value=lambda reposdir: packages)
+    monkeypatch.setattr(
+        os,
+        "environ",
+        {"CONVERT2RHEL_PACKAGE_NOT_UP_TO_DATE_SKIP": 1},
+    )
+    expected = set(
+        (
+            actions.ActionMessage(
+                level="WARNING",
+                id="PACKAGE_NOT_UP_TO_DATE_MESSAGE",
+                message=(
+                    "The system has 2 package(s) not updated based on the enabled system repositories.\n"
+                    "List of packages to update: package-1 package-2.\n\n"
+                    "Not updating the packages may cause the conversion to fail.\n"
+                    "Consider updating the packages before proceeding with the conversion."
+                ),
+            ),
+            actions.ActionMessage(
+                level="WARNING",
+                id="SKIP_PACKAGE_NOT_UP_TO_DATE",
+                message=(
+                    "Detected 'CONVERT2RHEL_PACKAGE_NOT_UP_TO_DATE_SKIP' environment variable, we will skip "
+                    "the package up-to-date check.\n"
+                    "Beware, this could leave your system in a broken state."
+                ),
+            ),
+        )
+    )
+
+    package_updates_action.run()
+    assert (
+        "The system has 2 package(s) not updated based on the enabled system repositories.\n"
+        in caplog.records[-1].message
+    )
+    assert expected.issuperset(package_updates_action.messages)
+    assert expected.issubset(package_updates_action.messages)
+
+
 def test_check_package_updates_with_repoerror(monkeypatch, caplog, package_updates_action):
-    get_total_packages_to_update_mock = mock.Mock(side_effect=pkgmanager.RepoError)
+    get_total_packages_to_update_mock = mock.Mock(side_effect=pkgmanager.RepoError("This is an error"))
+    monkeypatch.setattr(package_updates, "get_total_packages_to_update", value=get_total_packages_to_update_mock)
+    monkeypatch.setattr(package_updates, "get_total_packages_to_update", value=get_total_packages_to_update_mock)
+    package_updates_action.run()
+    print("start")
+    print(package_updates_action.result)
+    print("end")
```

Comment:
Don't forget to remove those debug statements here too! 


```suggestion
    print(package_updates_action.result)
```